### PR TITLE
Setup e2e tests for API

### LINKED ISSUE #3: [FEATURE] Scaffold HTTP tests
**Describe the solution you'd like**

Scaffold some HTTP tests to demonstrate how we can do lightweight integration-style tests in Go. The [httptest](https://pkg.go.dev/net/http/httptest) package provides facilities to execute requests over HTTP and assert on the responses.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by hasyimibhar**
Based on this issue: https://github.com/omshub/core-api/issues/3

This PR sets up lightweight tests for the API endpoints. The gin router is encapsulated inside an `api.Server`, which is just a thin wrapper of `http.Server`. `github.com/stretchr/testify` is added for testing convenience.

### CODE REVIEW

File: internal/api/base.go [hunk 1]
```diff
@@ -1,16 +1,55 @@
...
+	"time"
 
 	"github.com/gin-gonic/gin"
 )
 
-func Index(c *gin.Context) {
+type Server struct {
+	config     Config
+	router     *gin.Engine
+	httpServer *http.Server
```
Comment (disposedtrolley) [inline internal/api/base.go:14]: I like how you've decoupled the router from the server.

File: internal/api/config.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package api
+
+type Config struct {
```
Comment (disposedtrolley) [inline internal/api/config.go:3]: Nice one! This is going to be useful once we have more options.

File: internal/api/api_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+	resp, err := http.Get("http://localhost:3000/ping")
+	require.NoError(t, err)
+	defer resp.Body.Close()
+
+	require.Equal(t, 200, resp.StatusCode)
+
+	var response struct {
+		Message string `json:"message"`
+	}
```
Comment (disposedtrolley) [inline internal/api/api_test.go]: Thanks for going the extra mile and declaring a struct for the response.

File: cmd/api/main.go [hunk 4]
```diff
@@ -2,14 +2,12 @@ package main
...
 func main() {
-	router := gin.Default()
-	router.GET("/", api.Index)
-	router.GET("/ping", api.Ping)
+	server := api.NewServer(api.Config{
+		Port: "3000",
+	})
 
-	_ = router.Run(":3000")
+	_ = server.Serve()
```
Comment (disposedtrolley) [inline cmd/api/main.go:12]: We should probably dispatch a goroutine here to listen for OS signals, and invoke the `Shutdown` method when necessary. I'll make another issue for it, feel free to pick it up if you like!

File: internal/api/api_test.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package api
```
Comment (disposedtrolley) [inline internal/api/api_test.go]: Could we rename this to `base_test.go` after the file it's testing?

**Review by disposedtrolley — CHANGES_REQUESTED** [review]
This looks great! Thanks for hopping in and getting this done 🥳 

I'm happy to tick once `api_test.go` has been renamed.

File: cmd/api/main.go (hunk 4 above)
Comment (disposedtrolley) [reply cmd/api/main.go:12]: Done: https://github.com/omshub/core-api/issues/6

**Comment by disposedtrolley** [comment]
Also FYI, remember to rebase on https://github.com/omshub/core-api/pull/10 once it's merged to update the port number!

File: internal/api/api_test.go (hunk 5 above)
Comment (hasyimibhar) [reply internal/api/api_test.go]: Done.

**Review by disposedtrolley — APPROVED** [review]

**Review by premk — APPROVED** [review]
Good to go

**Pull request merged**